File: JavaStackQueue/edu/duq/mathcs/COSC435/Stack.java

/**
 * Represents a generic stack.
 * Implemented using the SingleLinkedList generic class.
 */
package edu.duq.mathcs.COSC435; 

public class Stack<T> 
{
    /**
     * A stack is represented using a SingleLinkedList.
     */
    private SingleLinkedList<T> stack = new SingleLinkedList<T>();

    // IMPLEMENT BELOW GENERIC PUBLIC METHODS WITH THE FOLLOWING NAMES
    //   push
	public void push(T pu) {
		stack.insertAtHead(pu); 
	}
    //   pop
	public T pop() {
		return stack.removeAtHead(); 
	}
    
}
    
    

File: C++StackQueue/stack.cpp
#include "stack.h"

  void stack::push(int value) {
    // Simulate push using this object's linked list
	s.insert_at_head(value); 
  }
  int stack::pop() {
    // Simulate pop using this object's linked list
	return s.remove_at_head(); 
  }

